src/lua/CommunityBalanceMod/CatpackBuffs/Exosuit.lua

```diff
@@ -0,0 +1,22 @@
+
+
+function Exosuit:OnWeldOverride(doer, elapsedTime, weldPerSecOverride)
+
+    -- macs weld marines by only 50% of the rate
+    local macMod = (HasMixin(self, "Combat") and self:GetIsInCombat()) and 0.1 or 0.5    
+    local weldMod = ( doer ~= nil and doer:isa("MAC") ) and macMod or 1
+
+    --balance mod
+    local catpackBonus = 1
+    if doer ~= nil and doer:GetParent() ~= nil and doer:GetParent():isa("Marine") and doer:GetParent():GetHasCatPackBoost() then 
```

Comment:
extra doer:GetParent() into a local variable and reuse it, there are 3 calls to getParent